src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlockMeasureCache.cs

```diff
@@ -66,20 +72,21 @@ public void CacheMeasure(TextBlock source, Size availableSize, Size measuredSize
 		{
 			var key = new MeasureKey(source);
 
-			if (!_entries.TryGetValue(key, out var entry))
-			{
-				Scavenge();
-				var node = _queue.AddLast(key);
-				_entries[key] = entry = new MeasureEntry(node);
-			}
-			else
+			if (_entries.TryGetValue(key, out var entry))
 			{
-				if (_queue.Count != 0 && _queue.Last.Value != key)
+				if (_queue.Count > 1 && _queue.Last.Value.Equals(key))
```

Comment:
You check if the `_queue.Last.Value.Equals(key)`, then if so you remove it and `AddLast` ... sounds weird

```suggestion
				if (_queue.Count > 1 && !_queue.Last.Value.Equals(key))
```